modules/deploy/pages/deployment-option/cloud/gcp-private-service-connect.adoc

```diff
@@ -0,0 +1,425 @@
+
+After <<get-a-cloud-api-access-token,getting an access token>>, you can <<create-new-byovpc-cluster-with-psc-enabled,enable PSC when creating a new BYOC cluster>>, or you can <<enable-psc-on-an-existing-byovpc-cluster,enable PSC for existing BYOC clusters>>.
+
+== Requirements
+
+* You must use the Redpanda Cloud API to enable the Redpanda endpoint service for your clusters. Follow the steps below to <<get-a-cloud-api-access-token,get an access token>>. 
+
+== Get a Cloud API access token 
+
+. Save the base URL of the Redpanda Cloud API in an environment variable:
++
+[,bash]
+----
+PUBLIC_API_ENDPOINT="https://api.cloud.redpanda.com"
+----
+
+. In the https://cloudv2.redpanda.com/[Redpanda Cloud UI], go to **Namespaces** and select the namespace in which you want to create a cluster.
++
+Copy and store the namespace ID (UUID) from the URL in the browser.
++
+[,bash]
+----
+NAMESPACE_ID=<uuid>
+----
+
+. Go to the organization (org) level **Home** in the UI, and navigate to **Clients**. If you don't have an existing client, you can create a new one by clicking **Add client**.
++
+Copy and store the client ID and secret.
++
+[,bash]
+----
+CLOUD_CLIENT_ID=<client-ID>
+CLOUD_CLIENT_SECRET=<client-secret>
+----
+
+. Get an API token using the client ID and secret. You can click the **Request an API token** link to see code examples to generate the token.
++
+[,bash]
+----
+AUTH_TOKEN=`curl -s --request POST \
+    --url 'https://auth.prd.cloud.redpanda.com/oauth/token' \
+    --header 'content-type: application/x-www-form-urlencoded' \
+    --data grant_type=client_credentials \
+    --data client_id=$CLOUD_CLIENT_ID \
+    --data client_secret=$CLOUD_CLIENT_SECRET \
+    --data audience=cloudv2-production.redpanda.cloud | jq .access_token | sed 's/"//g'`
+----
+
+You must send the API token in the `Authorization` header when making requests to the Cloud API.
+
+== BYOVPC
+
+You must use one additional customer-managed resource to use PSC on a BYOVPC Redpanda Cloud cluster. You need a NAT subnet with the *Purpose* set to `PRIVATE_SERVICE_CONNECT`. You can create the subnet using the `1_subnet` Terraform module. Review the module, then run the commands below.
+
+Make sure to provide your own values for the following:
+
+- `region`: The region of the Redpanda Cloud cluster.
+- `project_id`: The *host* GCP project ID.
+- `vpc_id`: The Redpanda Cloud cluster VPC ID.
+- `psc_nat_subnet_name`: The name of the NAT subnet for Private Service Connect.
+- `psc_nat_subnet_cidr`: The CIDR of the PSC NAT subnet. The mask should be at least `/29`. Each PSC connection takes up one IP address from the NAT subnet, so the CIDR must be able to accommodate all projects from which connections to the service attachment will be issued.
+
+See the official documentation for https://cloud.google.com/vpc/docs/configure-private-service-connect-producer#add-subnet-psc[creating a subnet for Private Service Connect^].
+
+[,bash]
+----
+cd terraform/1_subnet
```

Comment:
Is this `1_subnet` Terraform module an officially published module? I noticed that these commands imply navigating around the file structure here: https://github.com/redpanda-data/cloudv2/pull/12933/files Do we need to provide these files to our customers as well?

Reply:
These are modules I wrote because I didn't really knew how these docs worked. I saw https://github.com/redpanda-data/cloudv2/tree/main/tools/aws-private-link and tried to mirror that approach, but I don't know if it's the right one.
We could change this to a `gcloud` command instead if it's better.